src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,9 +974,52 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
+
+/*        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            String groupName = "";
+            if (!getCnFromAttributes) {
+                groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            } else {
+                for (NamingEnumeration ae = attributes.getAll(); ae.hasMore() && ae != null;) {
+                    Attribute attr = (Attribute) ae.next();
+                    if (CN.equals(attr.getID())) {
+                        for (NamingEnumeration e = attr.getAll(); groupName.equals("");e.hasMore()) {
+                            groupName = e.next().toString();
+                            if (e.hasMore()) {
+                                LOGGER.log(Level.FINE,"The group " + name.getRdns() + " has more than one cn value. The first one  (" + groupName + ") has been assigned as external group name");
+                            }
+                        }
+                    }
+                }
+            }
+            return groupName;
+        }*/
+
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            boolean isCN = false;
+            String groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            for (NamingEnumeration ae = attributes.getAll(); ae.hasMore();) {
+                Attribute attr = (Attribute) ae.next();
+                if (CN.equals(attr.getID())) {
+                    for (NamingEnumeration e = attr.getAll(); e.hasMore()&&!isCN;) {
```

Comment:
```suggestion
                    for (NamingEnumeration e = attr.getAll(); e.hasMore() && !isCN;) {
```

Reply:
So this will pick the last string in the list?
Why not just pick the first and not have ti loop?

Reply:
Is it just to be able to log the warning perhaps?